src/wolfsftp.c

```diff
@@ -1126,22 +1126,6 @@ int wolfSSH_SFTP_accept(WOLFSSH* ssh)
     if (ssh->error == WS_WANT_READ || ssh->error == WS_WANT_WRITE)
         ssh->error = WS_SUCCESS;
 
-    /* check accept is done, if not call wolfSSH accept */
-    if (ssh->acceptState < ACCEPT_CLIENT_SESSION_ESTABLISHED) {
```

Comment:
I like having the easy one API call for SFTP accept. If removing it though should we also change the wolfSSH_SFTP_connect to match?

Reply:
I would like the client-connect side to match the way the server-accept side works at some point.

I believe the SFTP accept needs to happen after the handshake is complete. The wolfSSH_accept() should only do KEX and user authentication. After userauth, any sort of channel can be open. In fact, channel open doesn't specify the kind of channel. That's all channel requests when the channel has been allocated.

I guess an alternative could be to have wolfSSH_SFTP_accept() call wolfSSH_accept() first thing and if the SSH connection is established and authenticated, return success. Then roll into doing the SFTP accept actions. I don't think SFTP should know about the WOLFSSH's internal state and where its handshake is at.

Reply:
marking as resolved, we will need to update the documentation and add the change behavior to the next release. Having the clean boundary between internal SSH operations and post handshake operations is nice though.

Reply:
I'm thinking this is a v1.5 thing. And I agree with the documentation update. I have another branch that works off this to make the servers even better.